doc/record-encryption.md

```diff
@@ -0,0 +1,55 @@
+Record Encryption
+=================
+
+This covers the technique used for encrypting records and the reasoning behind the choices.  Configuration of the encryption methods is [covered separately][encryption config].
+
+All data is stored in the [storage engine] encrypted.  Twice.  We use a layered encryption scheme to safeguard all data.  This is in addition to any encryption that is offered by the storage engine, such as how S3 records are encrypted when saved.
+
+
+Layered Encryption
+------------------
+
+The technique of layered encryption is where you take some information and encrypt it multiple times.  You can provide the encryption keys to separate parties in order to keep the encrypted data more secure.  A different approach is to use a single secret that you build from enough information from various people; see [Shamir's Secret Sharing](https://en.wikipedia.org/wiki/Shamir%27s_Secret_Sharing).
+
+Layered encryption provides benefits.  For instance, PCI DSS 3.2 has some rules governing key rotation in section 3.6 that could apply.  The service is able to decrypt the outer layer and rekey the stored information.  Also, some information may be acceptable to be seen again without decrypting the entire record.  As an example, the account record that is stored holds the MFA key and hashed password in the inner layer to be safe.  The outer layer encrypts the email address, so emails to the user may be sent before their password expires.
```

Comment:
Can't link.  It's behind a non-distribution agreement.

Disagreed with the comma.  Preferred rewording of the sentence.